File: 0009-palindrome-number/0009-palindrome-number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        value = str(x)

        left = 0
        right = len(value) - 1
        result = True

        while (left < right):
            if value[left] != value[right]:
                result = False
                break
            
            left += 1
            right -= 1
        
        return result
        

File: LeetCode/Unknown/0009-palindrome-number/0009-palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        if (x < 0) {
            return false;
        }

        String value = String.valueOf(x);

        int left = 0;
        int right = value.length() - 1;
        boolean result = true;

        while (left < right) {
            if (value.charAt(left) != value.charAt(right)) {
                result = false;
                break;
            }

            left++;
            right--;
        }

        return result;
    }
}